### PR TITLE
UI Bugfix Mobile: Add distance at the bottom, if there is a difference between `window.innerHeight` and `window.outerHeight` 

### PR DESCRIPTION
**Pull request by relativisticelectron**
Fix for https://github.com/cryptoadvance/specter-desktop/issues/1826

### CODE REVIEW

**Comment by moneymanolis** [comment]
This currently creates huge additional space at the bottom of the tx table for me:

grafik

I think it would be good to use media queries to restrict the extra spacing to mobile devices, have a look here for example: 
https://wpbuffs.com/media-query-for-mobile/

I could not reproduce the issue in the first place using Firefox but this depends on so much stuff besides the browser.

File: src/cryptoadvance/specter/static/styles.css [hunk 1]
```diff
@@ -1244,3 +1244,16 @@ input:checked + .slider:before {
...
     transform: translateX(26px);
 }
+
+
+
+/* The viewable height is often smaller than the height given by the browser. 
+See: https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
+This style is used for a div at the bottom in the <main> element to create a distance at the bottom of the pages
+*/
+.mobile-distance-element-botom {
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/static/styles.css:1254]: Typo, should be bot**t**om

File: src/cryptoadvance/specter/templates/device/device.jinja [hunk 2]
```diff
@@ -138,33 +138,34 @@
...
-			<button type="submit" style="background: none; color: inherit; border: none; padding: 0; font: inherit; cursor: pointer; outline: inherit;">
-				<div class="small-card radio">
-					<img src="{{ url_for('static', filename='img/key.svg') }}" class="svg-white" style="width: 120px;">
-					{{ _("Create single key wallet") }}
-				</div>
-			</button>
-		</form>
-		{% if specter.device_manager.devices | length > 1 %}
-			<a href="{{ url_for('wallets_endpoint.new_wallet', wallet_type='multisig') }}" style="text-decoration: none;">
+		<h1>{{ _("New device was added successfully!") }}</h1>
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/device/device.jinja:141]: Looks like GitHub is showing a weird diff here, this code is already on master as far as I can see ...

File: src/cryptoadvance/specter/static/styles.css [hunk 3]
```diff
@@ -1244,3 +1244,16 @@ input:checked + .slider:before {
...
+    height: mobileDistanceElementBotomHeight; /* Fallback for browsers that do not support Custom Properties */
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/static/styles.css:1255]: How is this a fallback? What would such a browser do with "mobileDistanceElementBotomHeight"?

**Comment by relativisticelectron** [comment]
> This currently creates huge additional space at the bottom of the tx table for me:

Strange. Ok. Then this difference calculation is not a good idea.

> 
> I think it would be good to use media queries to restrict the extra spacing to mobile devices, have a look here for more on that: https://wpbuffs.com/media-query-for-mobile/
> 
> Sth. like this:
> 
> ```
> 
> @media only screen and (max-width: 600px) {
> .mobile-distance-element-botom {
>     height: mobileDistanceElementBotomHeight; /* Fallback for browsers that do not support Custom Properties */
>     height: var(--mobileDistanceElementBotomHeight, mobileDistanceElementBotomHeight);
>     flex-shrink: 0;
>     width: 100%;
>   }
> }
> ```
> 
> BTW: I could not reproduce the issue in the first place using Firefox but this depends on so much stuff besides the browser. Update: Could reproduce it with Safari.

Ok, I will try the `@media`.

**Comment by moneymanolis** [comment]
Also: Did you try this approach as well? 
https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
Would be a bit more tailored towards the individual viewports than the difference between `innerHeight` and `outerHeight`.

Does not work for Safari for me, though, and Firefox works fine without anything for me, so can't verify whether this helps as well.

File: src/cryptoadvance/specter/templates/device/device.jinja (hunk 2 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/templates/device/device.jinja:141]: There was a `/div`  missing on master, and the inner codeblock was not indented.

**Comment by relativisticelectron** [comment]
> Also: Did you try this approach as well? https://css-tricks.com/the-trick-to-viewport-units-on-mobile/ Would be a bit more tailored towards the individual viewports than the difference between `innerHeight` and `outerHeight`.
> 
> Does not work for Safari for me, though, and Firefox works fine without anything for me, so can't verify whether this helps as well.

This is where I have the idea from to use the `innerHeight`, and then discovered that `outerHeight - innerHeight` fits the height of the navigation bar.  Their JS code is not directly applicable, because we have nothing to shrink. On our  endpoints are various elements that get cut off by the navigation bar.

**Comment by relativisticelectron** [comment]
The `window.outerHeight` was the wrong [approach](https://developer.mozilla.org/en-US/docs/Web/API/Window/outerHeight).   
I now changed it to a fixed height if it is a narrow screen.

File: src/cryptoadvance/specter/static/styles.css [hunk 4]
```diff
@@ -1249,6 +1264,10 @@ input:checked + .slider:before {
     -ms-transform: translateX(26px);
     transform: translateX(26px);
 }
+
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/static/styles.css:1267]: Why those extra lines? Remove?

File: src/cryptoadvance/specter/static/styles.css [hunk 5]
```diff
@@ -1019,6 +1019,18 @@ nav.collapse-on-mobile a.mobile-nav-icon {
     display: none;
 }
 
+
+/* The viewable height is often smaller than the height given by the browser. 
+See: https://css-tricks.com/the-trick-to-viewport-units-on-mobile/
+This style is used for a div at the bottom in the <main> element to create a distance at the bottom of the pages
+*/
+.mobile-distance-element-botom {
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/static/styles.css:1027]: Please fix the spelling of bottom

File: src/cryptoadvance/specter/static/styles.css [hunk 6]
```diff
@@ -1192,6 +1204,9 @@ nav.collapse-on-mobile a.mobile-nav-icon {
     .main-page-container{
         background-image: none;
       }
+    .mobile-distance-element-botom {
+        height: var(--mobileDistanceElementBotomHeight, mobileDistanceElementBotomHeight);
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/static/styles.css:1208]: Doesn't make sense to repeat the variable here, better to provide some px value as fallback.

grafik

File: src/cryptoadvance/specter/static/styles.css (hunk 4 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/static/styles.css:1267]: done

File: src/cryptoadvance/specter/static/styles.css (hunk 5 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/static/styles.css]: Sorry, I didn't even see the typo until now!   Done.

File: src/cryptoadvance/specter/static/styles.css (hunk 6 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/static/styles.css]: Thanks for the reminder. Done.

File: src/cryptoadvance/specter/static/styles.css (hunk 3 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/static/styles.css]: I removed it.

File: src/cryptoadvance/specter/static/styles.css (hunk 1 above)
Comment (relativisticelectron) [reply src/cryptoadvance/specter/static/styles.css]: Done. (now)

File: src/cryptoadvance/specter/templates/base.jinja [hunk 7]
```diff
@@ -40,6 +40,9 @@
 			}
 		</style>
 		<script type="text/javascript" src="{{ url_for('static', filename='helpers.js') }}"></script>
+		<script>
```
Comment (moneymanolis) [inline src/cryptoadvance/specter/templates/base.jinja:43]: Can we move this to `helpers.js`?

Comment (relativisticelectron) [reply src/cryptoadvance/specter/templates/base.jinja]: Yes, that fits well in there. Done.

Comment (moneymanolis) [reply src/cryptoadvance/specter/templates/base.jinja]: Great, thanks.

**Review by moneymanolis — APPROVED** [review]

**Pull request merged**